File: HarvestHub/Models/Category.cs
using System.ComponentModel.DataAnnotations;
using HarvestHub.Models;

public class Category
{
    [Key]
    public int CategoryId { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }

    public int? ParentId { get; set; }
    public Category Parent { get; set; }
    public ICollection<Category> Children { get; set; }
    public ICollection<Product> Products { get; set; }
}

File: harvesthub-frontend/src/types/category.ts
export interface Category {
    id: string;
    name: string;
    description?: string;
    imageUrl?: string;
    parentId?: string;
    children?: Category[];
    productsCount: number;
}

export interface ProductFilter {
    categoryId?: string;
    search?: string;
    minPrice?: number;
    maxPrice?: number;
    isOrganic?: boolean;
    farmerId?: string;
    page?: number;
    pageSize?: number;
    sortBy?: 'name' | 'price' | 'rating' | 'createdAt';
    sortOrder?: 'asc' | 'desc';
}